src/Parser/Monk/Brewmaster/Modules/Core/StaggerFabricator.js

```diff
@@ -0,0 +1,104 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import { GIFT_OF_THE_OX_SPELLS } from '../../Constants';
+
+const PURIFY_BASE = 0.4;
+const ELUSIVE_DANCE_PURIFY = 0.2;
+const STAGGERING_AROUND_PURIFY = 0.01;
+const ISB_QUICK_SIP_PURIFY = 0.05;
+const T20_4PC_PURIFY = 0.05;
+const debug = false;
+
+export const EVENT_STAGGER_POOL_UPDATE = "stagger_pool_update";
+export const KIND_STAGGER_ADDED = "added";
+export const KIND_STAGGER_REMOVED = "removed";
+
+/**
+ * Fabricate events corresponding to stagger pool updates. Each stagger
+ * tick, purify, and stagger absorb generates one event.
+ */
+class StaggerFabricator extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  }
+
+  _has_t20_4pc = false;
+  _PURIFY_AMOUNT = PURIFY_BASE;
+  _stagger_pool = 0;
+  _has_quick_sip = false;
+
+  get staggerPool() {
+    return this._stagger_pool;
+  }
+
+  on_initialized() {
+    const player = this.combatants.selected;
+    if(player.hasTalent(SPELLS.ELUSIVE_DANCE_TALENT.id)) {
+      this._PURIFY_AMOUNT += ELUSIVE_DANCE_PURIFY;
+    }
+    this._PURIFY_AMOUNT += player.traitsBySpellId[SPELLS.STAGGERING_AROUND.id] * STAGGERING_AROUND_PURIFY;
+    this._has_quick_sip = player.traitsBySpellId[SPELLS.QUICK_SIP.id] > 0;
+    this._has_t20_4pc = player.hasBuff(SPELLS.XUENS_BATTLEGEAR_4_PIECE_BUFF_BRM.id);
+  }
+
+  on_toPlayer_absorbed(event) {
+    if (event.ability.guid === SPELLS.STAGGER.id) {
```

Comment:
It's preferred to avoid nesting if-statements. This could also be written like so at the top of the method:

```jsx
if (event.ability.guid !== SPELLS.STAGGER.id) {
  return;
}
```